http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -204,15 +204,6 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
         } yield {
           savedCluster
         }
-
-        // If cluster creation failed on the Google side, createGoogleCluster removes resources in Google.
```

Comment:
I think this logic was just a mistake. We should treat an `Error` cluster as an active Sam resource, since you can still perform operations on it. The only time we call `notifyClusterDeleted` now is when the cluster transitions to `Deleted` state in `ClusterMonitorActor`.

Reply:
I did a little spelunking to help figure out why we had this logic.  I wasn't successful, but maybe these links will be educational: #137 #139 #167

Reply:
Thanks, that's useful. I think we were on the right track but missed this subtlety.